### PR TITLE
Issue 489: Fixed es_indices() to return a matching index

### PR DESCRIPTION
**Pull request by vichargrave**
closes: #489

### CODE REVIEW

**Comment by jxstanford** [comment]
could you update/create tests for reasonable positive/negative cases for this function? (empty string, complete index name, wildcard)...

**Comment by vichargrave** [comment]
I updated the test_es_indices() to call the es_indices() with an index prefix that contains "*" at the end of the prefix string which is what the prefixes contain when running the application.  For example, when querying for API status, the index prefix is "api_stats-*", so es_indices() must strip the "*" so that the mapping for the api_stats-yyyy-mm-dd index can be retrieved from Elasticsearch.

File: goldstone/tests.py [hunk 1]
```diff
@@ -166,12 +166,12 @@ def test_es_indices(self, m_conn):
...
         # test with no conn
-        result = es_indices(prefix='index')
+        result = es_indices(prefix='index1*')
         self.assertTrue(m_es.indices.status.called)
         self.assertIn('index1', result)
         self.assertNotIn('not_index1', result)
 
         # test with prefix
-        result = es_indices('index', es_conn())
+        result = es_indices('index1*', es_conn())
```
Comment (jxstanford) [inline goldstone/tests.py]: would probably be better coverage to test a couple different conditions.  so, in this series of tests, we should have ones that test a string without a * a string with a *, and anything else that we might see in the real world.

Comment (jxstanford) [inline goldstone/tests.py]: in other words, leave the old test, rather than replace it, and verify that we get what we expect.

**Comment by vichargrave** [comment]
Excellent point.  I restored the original tests and added additional ones.

**Comment by jxstanford** [comment]
confirmed that exceptions are gone locally, as long as circleci passes, 👍

**Comment by vichargrave** [comment]
https://github.com/Solinea/goldstone-server/issues/489

**Pull request merged**